tools/picard.wdl

```diff
@@ -865,3 +865,180 @@ task bam_to_fastq {
+            choices: [
+                'BALANCING_WITHOUT_INTERVAL_SUBDIVISION_WITH_OVERFLOW',
+                'INTERVAL_SUBDIVISION',
+                'BALANCING_WITHOUT_INTERVAL_SUBDIVISION'
+            ]
+        }
+        unique: "Should the output interval lists contain unique intervals? Implies sort=true. Merges overlapping or adjacent intervals."
+        sort: "Should the output interval lists be sorted? Sorts by coordinate."
+        scatter_count: "Number of interval lists to create"
+        modify_memory_gb: "Add to or subtract from dynamic memory allocation. Default memory is determined by the size of the inputs. Specified in GB."
+    }
+
+    input {
+        File interval_list
+        String subdivision_mode = "BALANCING_WITHOUT_INTERVAL_SUBDIVISION_WITH_OVERFLOW"
+        Boolean unique = true
+        Boolean sort = true
+        Int scatter_count
+        Int modify_disk_size_gb = 0
+    }
+
+    Int disk_size_gb = 1 + modify_disk_size_gb
+
+    command <<<
+        set -euo pipefail
+
+        mkdir out
+        picard -Xms1g \
+            IntervalListTools \
+            SCATTER_COUNT=~{scatter_count} \
+            SUBDIVISION_MODE=~{subdivision_mode} \
+            UNIQUE=~{unique} \
+            SORT=~{sort} \
+            INPUT=~{interval_list} \
+            OUTPUT=out
+
+        bash <<CODE
+        I=0
+        for list in \$(ls out/*/*.interval_list)
+        do
+           I=\$((I+1))
+           dir=\$(dirname \$list)
+           name=\$(basename \$list)
+           mv \$list \${dir}/\${I}\${name}
+        done
+        echo \$I > interval_count.txt
+        CODE
+    >>>
+
+    output {
+        Array[File] out = glob("out/*/*.interval_list")
+        Int interval_count = read_int("interval_count.txt")
+    }
+
+    runtime {
+        memory: "2 GB"
+        disk: "~{disk_size_gb} GB"
+        container: "quay.io/biocontainers/picard:2.27.5--hdfd78af_0"
+        maxRetries: 1
+    }
+}
+
+task create_sequence_dictionary {
+    meta {
+        description: "Creates a sequence dictionary for the input FASTA file using Picard"
+        external_help: "https://gatk.broadinstitute.org/hc/en-us/articles/13832748622491-CreateSequenceDictionary-Picard-"
+        outputs: {
+            dictionary: "Sequence dictionary produced by `picard CreateSequenceDictionary`."
+        }
+    }
+
+    parameter_meta {
+        fasta: "Input FASTA format file from which to create dictionary"
+        outfile_name: "Name for the CreateSequenceDictionary dictionary file"
+        memory_gb: "RAM to allocate for task, specified in GB"
+        modify_disk_size_gb: "Add to or subtract from dynamic disk space allocation. Default disk size is determined by the size of the inputs. Specified in GB."
+    }
+
+    input {
+        File fasta
+        String outfile_name = basename(fasta, ".fa") + ".dict"
+        String? assembly_name
+        String? fasta_url
+        String? species
+        Int memory_gb = 16
```

Comment:
Can just put `memory: "16 GB"` directly in the runtime block instead of this.

Reply:
I use that value to compute `java_heap_size` which gets passed to the JVM. If WDL gets support for accessing runtime values in the `command` block eventually, we can remove this.

Reply:
Ah yes. We've used this elsewhere too. Looks good then 👍